### PR TITLE
BFD-1719: Fix pipeline-grpc issues and enable checkstyle

### PR DESCRIPTION
**Pull request by lsmitchell**
**JIRA Ticket:**
[BFD-1719](https://jira.cms.gov/browse/BFD-1719)

**User Story or Bug Summary:**

As a BFD Engineer I would like to have an automated checkstyle process in place for the BFD javadocs so that the code is documented, automatically checked for static javadoc issues, and reviewers can focus on elements of code that need human focus.

---

### What Does This PR Do?

Fixes the 1000+ checkstyle issues in pipeline-grpc and enables checkstyle for this module.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] The data dictionary has been updated with any field mapping changes, if any were made.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/source/ClaimTransformerFieldTester.java [hunk 1]
```diff
@@ -44,7 +44,6 @@
    *     entity object
    * @param fieldLabel text identifying the field in {@link DataTransformer.TransformationException
    *     error messages}
-   * @param minLength minimum valid length for the string field
```
Comment (mjburling) [inline apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/source/ClaimTransformerFieldTester.java:47]: Interesting to see the `minLength` dropped out of these!

**Review by mjburling — APPROVED** [review]
Sometimes, I do question the value with lots of e.g.

```java
/**
 * Frobnicates foo.
 *
 * @param foo the foo to frobnicate
 * @return a cromulant foo via frobnication
 */
public fooFrobnicator(String foo) {
// ...
}
```

I mean... not _really_ but you get the idea. I'm trusting the process and want to see this through. I've left a handful of minor comments/suggestions but overall... nothing shocking. I'm lending my approval anticipating a review re-request.

Thanks, @lsmitchell

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/source/ClaimTransformerFieldTester.java (hunk 1 above)
Comment (lsmitchell) [reply apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/source/ClaimTransformerFieldTester.java:47]: Was removed from the test at some point at the test, this is just fixing the docs to match. Not sure behind the actual reason for the original adjustment though.

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/source/McsClaimTransformerTest.java [hunk 2]
```diff
@@ -1519,101 +1977,132 @@ RdaChange<RdaMcsClaim> transformClaim(McsClaim claim) {
...
     @Override
     String getLabel(String basicLabel) {
       return "adjustments-0-" + basicLabel;
     }
   }
 
+  /**
+   * Adaptor class extending the {@link ClaimTransformerFieldTester} class that can be used to
+   * create {@link McsAudit.Builder} instances and to trigger a transformation of a claim. Used for
+   * tests that operator on {@link McsAudit} and {@link RdaMcsAudit} instances.
```
[the remark this answers is not in the document]
Comment (lsmitchell) [reply apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/source/McsClaimTransformerTest.java:2039]: Accursed copy/paste issue. Fixed.

**Comment by lsmitchell** [comment]
> Sometimes, I do question the value ...
> 

Yeah some docs end up of less value when there is nothing much to say, but IMO the only way to make people think about keeping documentation on things that _really_ need it is to consistently require something be there. (Plus for some things like constructors the _description_ may be of less value, but the parameters of much more value; however the tooling doesnt allow us to pick and choose.)

**Review by mjburling — APPROVED** [review]

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/server/S3JsonMessageSource.java [hunk 3]
```diff
@@ -20,11 +20,21 @@
  * @param <T> the type parameter
  */
 public class S3JsonMessageSource<T> implements MessageSource<T> {
+  /** The S3 object to read. */
```
Comment (mackec1) [inline apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/server/S3JsonMessageSource.java:23]: This is nice; provides a nice operational model for creating a S3RifMessageSource to stream RIF files for ETL.

**Review by mackec1 — APPROVED** [review]
First off...a herculean effort to do this...bravo. Also, my mind must have wandered as I kept thinking it was just regular PACA code by either Brian or Kyle so ignore comments that talked about leveraging some of their stuff for BFD pipeline.

**Pull request merged**